contrib/opencensus-ext-azure/opencensus/ext/azure/common/transport.py

```diff
@@ -102,7 +102,7 @@ def _transmit(self, envelopes):
                                 error['message'],
                                 envelopes[error['index']],
                             )
-                    if resend_envelopes:
+                    if resend_envelopes and self.storage is not None:
```

Comment:
This seems to be hack, need to clean up.

Reply:
I refactored the "rules" for what _transmit will return and introduced a "NoopStorage" that is used for metrics exporter.